wom/src/main/scala/wdl4s/wdl/AstTools.scala

```diff
@@ -306,7 +306,7 @@ object AstTools {
         && a.getAttribute("rhs").isTerminal => a.getAttribute("rhs").asInstanceOf[Terminal]
     }
 
-    def fullVariableReferenceString = terminal.getSourceString + terminalSubIdentifier.map("." + _.getSourceString).getOrElse("")
+    def fullVariableReferenceString = terminal.getSourceString + terminalSubIdentifier.fold(""){ "." + _.getSourceString }
```

Comment:
😞  It's shorter and typesafe.  I don't think unfamiliarity is a valid argument, especially as this use case is not obscure and thus `fold` could be used regularly.